x-pack/elastic-agent/pkg/fleetapi/client/client.go

```diff
@@ -36,7 +38,7 @@ type Sender interface {
 
 var baseRoundTrippers = func(rt http.RoundTripper) (http.RoundTripper, error) {
 	rt = NewFleetUserAgentRoundTripper(rt, release.Version())
-	return rt, nil
+	return apmhttp.WrapRoundTripper(rt), nil
```

Comment:
right, this will propagate the trace with the correct tracing headers :) I haven't tried it yet, actually, but [fleet-server has been instrumented](https://github.com/elastic/fleet-server/pull/898) so it should work with a build based off this branch